x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueQuery.java

```diff
@@ -107,7 +107,15 @@ public static class Builder extends AbstractQueryBuilder<Builder> {
             this.next = in.readNamedWriteable(QueryBuilder.class);
             this.field = in.readString();
             if (in.getTransportVersion().onOrAfter(TransportVersions.ESQL_SINGLE_VALUE_QUERY_SOURCE)) {
-                this.source = Source.readFrom((PlanStreamInput) in);
+                if (in instanceof PlanStreamInput psi) {
+                    this.source = Source.readFrom(psi);
+                } else {
+                    /*
+                     * For things like CanMatch we serialize without the Source. But we
+                     * don't use it, so that's ok.
+                     */
+                    this.source = Source.readEmpty(in);
```

Comment:
Thought: instead of an empty source, would it be reasonable to keep at least the line, column, and length around for debugging?

Reply:
Another (minor) idea could be to "normally" `readFrom()` and then just `this.source = Source.EMPTY;` here, rather than make SVG's need Source's concern.
But either way is 👍 .